specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -0,0 +1,383 @@
+  ) => BackendStat['value']
+): StatCategoryReturn {
+  if (backendobject === undefined) {
+    return undefined;
+  }
+  return Object.fromEntries(
+    Object.entries(backendobject).map(([key, value]) => [
+      key,
+      {
+        label: key,
+        spec: {
+          type: 'Backendstat',
+          value: modifyfunction(value),
+        },
+      },
+    ])
+  );
+}
+
+export type StatCategoryReturn =
+  | IR<{ readonly label: string; readonly spec: StatItemSpec }>
+  | undefined;
+
+type StatsSpec =
+  | {
+      readonly [CATEGORY_NAME in keyof BackendStatsResult]: {
+        readonly label: string;
+        readonly categories: (
+          backendStatsResult: BackendStatsResult[CATEGORY_NAME] | undefined
+        ) => StatCategoryReturn;
+      };
+    }
+  | {
+      readonly [CATEGORY_NAME in string]: {
+        readonly label: string;
+        readonly categories: () => StatCategoryReturn;
+      };
+    };
+
+export const statsSpec: StatsSpec = {
+  holdings: {
+    label: 'Holdings',
+    categories(backendStatsResult) {
+      return {
+        specimen: {
+          label: 'Specimens',
+
+          spec: {
+            type: 'Querybuildstat',
+            tableName: 'CollectionObject',
+            fields: [],
+          },
+        },
+        preparation: {
+          label: 'Preparations',
+          spec: {
+            type: 'Querybuildstat',
+            tableName: 'Preparation',
+            fields: [{ position: 0, path: 'countAmt' }],
+          },
+        },
+        typeSpecimen: {
+          label: 'Type Specimens',
+          spec: {
+            type: 'Querybuildstat',
+            tableName: 'Determination',
+            fields: [
+              {
+                position: 0,
+                path: 'typeStatusName',
+                operStart: queryFieldFilters.equal.id,
+                isNot: true,
+              },
+            ],
+          },
+        },
+        familiesRepresented: {
+          label: 'Families Represented',
+          spec: {
+            type: 'Backendstat',
+            value: backendStatsResult?.familiesRepresented,
+          },
+        },
+        generaRepresented: {
+          label: 'Genera Represented',
+          spec: {
+            type: 'Backendstat',
+            value: backendStatsResult?.generaRepresented,
+          },
+        },
+        speciesRepresented: {
+          label: 'Species Represented',
+          spec: {
+            type: 'Backendstat',
+            value: backendStatsResult?.speciesRepresented,
+          },
+        },
+      };
+    },
+  },
+  preparation: {
+    label: 'Preparations',
+    categories: (backendStatsResult) =>
+      modifyBackendResult<'preparation'>(
+        backendStatsResult,
+        (prepelement) =>
+          `${formatNumber(prepelement.lots)} / ${formatNumber(
+            prepelement.total
+          )}`
+      ),
+  },
+  loans: {
+    label: 'Loans',
+    categories() {
```

Comment:
This return is unnecessary. Consider replacing
```js
categories() {
  return {
    ...
  }
}
```
with 
```js
categories: ()=>({
  ...
})
```

Same for the following functions in this file